RecyclerView/src/main/kotlin/com/infomaniak/core/recyclerview/ResettableLazy.kt

```diff
@@ -30,7 +30,8 @@ internal class ResettableLazy<T : Any>(private val initializer: () -> T) {
         return value ?: initializer().apply { value = this }
     }
 
-    fun invalidate() {
-        value = null
-    }
+    /**
+     * Returns the value prior to invalidation/removal.
+     */
+    fun invalidate(): T? = value.also { value = null }
```

Comment:
Also, I put this here because I just saw it but, ResettableLazy's method is called "invalidate" instead of "reste", I feel like we missed an opportunity there

Reply:
Renamed it to `reset`.